apis/run/v1alpha3/kappcontrollerconfig_types.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// KappControllerConfigSpec defines the desired state of KappControllerConfig
+type KappControllerConfigSpec struct {
+	// The namespace in which calico is deployed
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=kube-system
+	Namespace string `json:"namespace,omitempty"`
+
+	KappController KappController `json:"kappController,omitempty"`
+}
+
+type KappController struct {
+	// Whether to create namespace specified for kapp-controller
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=true
+	CreateNamespace bool `json:"createNamespace,omitempty"`
+
+	// The namespace value used for global packaging resources. Any Package and PackageMetadata CRs within that namespace will be included in all other namespaces on the cluster, without duplicating them
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=tanzu-package-repo-global
```

Comment:
I think we need to rethink this as kapp-controller doesn't support multiple global namespaces. Could we make this tkg-system? cc @imikushin 

Probably need input from TMC team when we change this.

Reply:
AI: Follow up on this

Reply:
Could we log an issue?

Reply:
https://github.com/vmware-tanzu/tanzu-framework/issues/1614